packages/design-system/src/molecules/header/index.tsx

```diff
@@ -1,12 +1,28 @@
 import { useBreakpointValue } from 'native-base'
+import { PropsWithChildren } from 'react'
+import { Platform, ScrollView, View } from 'react-native'
+import { MobileSidebar } from '../sidebar/mobile'
 import { DesktopHeader } from './desktop'
-import { MobileHeader } from './mobile'
 import { HeaderProps } from './types'
 
-export const HeaderV2 = (props: HeaderProps) => {
+export const HeaderV2: React.FC<HeaderProps & PropsWithChildren> = ({
+  children,
+  ...props
+}) => {
   const isMobile = useBreakpointValue({ base: true, md: false })
+  const isNative = Platform.OS !== 'web'
   if (isMobile) {
-    return <MobileHeader {...props} />
+    return (
+      <View>
+        <MobileSidebar {...props} options={[props.options]} />
+        {isNative ? <ScrollView>{children}</ScrollView> : children}
```

Comment:
We can avoid `isNative` variable declaration by using `Platform.select`.

```suggestion
        {Platform.select({ native: <ScrollView>{children}</ScrollView>, default: children })}
```